skimage/feature/peak.py

```diff
@@ -138,12 +139,16 @@ def peak_local_max(image, min_distance=10, threshold_abs=0, threshold_rel=0.1,
         # zero out the image borders
         for i in range(image.ndim):
             image = image.swapaxes(0, i)
-            image[:min_distance] = 0
-            image[-min_distance:] = 0
+            remove = (footprint.shape[i] if footprint is not None
+                      else 2 * min_distance)
+            image[:remove // 2] = 0
+            image[-remove // 2:] = 0
             image = image.swapaxes(0, i)
 
     # find top peak candidates above a threshold
-    peak_threshold = max(np.max(image.ravel()) * threshold_rel, threshold_abs)
+    peak_threshold = threshold_abs
+    if threshold_rel is not None:
```

Comment:
Instead of defaulting to `None`, you could, instead, set `threshold_rel=1` by default, and get rid of this conditional.